packages/discord-bot/src/utils/assertPraiseAllowedInChannel.ts

```diff
@@ -0,0 +1,38 @@
+import { CommandInteraction, TextBasedChannel } from 'discord.js';
+import { settingValue } from 'api/dist/shared/settings';
+
+const getChannelId = (channel: TextBasedChannel): string => {
+  return channel.type === 'GUILD_PUBLIC_THREAD' ||
+    channel.type === 'GUILD_PRIVATE_THREAD' ||
+    channel.type === 'GUILD_NEWS_THREAD'
+    ? channel?.parent?.id || channel.id
+    : channel.id;
+};
+
+export const assertPraiseAllowedInChannel = async (
+  interaction: CommandInteraction
+): Promise<boolean> => {
+  const { channel } = interaction;
+  const allowedInAllChannels = (await settingValue(
+    'PRAISE_ALLOWED_IN_ALL_CHANNELS'
+  )) as boolean;
+  const allowedChannelsList = (await settingValue(
+    'PRAISE_ALLOWED_CHANNEL_IDS'
+  )) as number[];
+  if (
+    !channel ||
+    (!allowedInAllChannels &&
+      !allowedChannelsList.includes(Number.parseInt(getChannelId(channel))))
+  ) {
+    await interaction.editReply(
+      `**❌ Praise Restricted**\nPraise not allowed in this channel.\nTo praise, use the following channels - ${allowedChannelsList
```

Comment:
The list of allowed channels is not being displayed when praising in a channel that is not allowed.
 
<img width="346" alt="Skärmavbild 2022-06-20 kl  13 03 13" src="https://user-images.githubusercontent.com/9698363/174588543-782744d7-6bff-4fe3-a5ac-966e9187d578.png">

Reply:
yes, that's a result of using not using strings for this.
The `number` type used by `IntegerList` can't store numbers of this length.